pixeltable/catalog/table_version_handle.py

```diff
@@ -76,6 +78,11 @@ class ColumnHandle:
     col_id: int
 
     def get(self) -> 'Column':
+        if self.col_id not in self.tbl_version.get().cols_by_id:
+            raise excs.Error(
+                f'Column has been dropped (no record for column ID {self.col_id} '
```

Comment:
You could actually tell the user at which version it was dropped, that's available in _tbl_md.column_md[col_id].schema_version_drop.